adapter/service/outlvlswitch/routing.go

```diff
@@ -41,6 +44,90 @@ func RouteCmdLvlSet(serviceRegistry adapter.ServiceRegistry) *router.Routing {
 	)
 }
 
+// RouteCmdLvlStart returns a routing responsible for handling the command.
+func RouteCmdLvlStart(serviceRegistry adapter.ServiceRegistry) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlStart(serviceRegistry),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlStart),
+	)
+}
+
+// RouteCmdLvlStop returns a routing responsible for handling the command.
+func RouteCmdLvlStop(serviceRegistry adapter.ServiceRegistry) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlStop(serviceRegistry),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlStop),
+	)
+}
+
+// HandleCmdLvlStart returns a handler responsible for handling CmdLvlStart message.
+func HandleCmdLvlStart(serviceRegistry adapter.ServiceRegistry) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := serviceRegistry.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			service, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			direction, err := message.Payload.GetStringValue()
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting level value from message")
+			}
+
+			duration, err := getDurationInSeconds(message)
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting duration value from message")
+			}
+
+			startLvl, err := getStartLvl(message)
+			if err != nil {
+				return nil, errors.Wrap(err, "adapter: error while getting start_lvl value from message")
+			}
+
+			if err := service.StartLevelTransition(direction, startLvl, duration); err != nil {
+				return nil, errors.Wrap(err, "adapter: failed to start level transitioning")
+			}
+
```

Comment:
Yes, we should send level report, because startLvl might be different than current level. For instance lvl is set to 20 but I request transition down from 80.

Reply:
@rstraczynski @viktor-viktor